src/main/java/io/datanerds/avropatch/Patch.java

```diff
@@ -49,14 +137,16 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) {
             return false;
         }
-        Patch patch = (Patch) o;
+        Patch<T> patch = (Patch<T>) o;
         return Objects.equals(headers, patch.headers) &&
+                Objects.equals(resource, patch.resource) &&
+                Objects.equals(timestamp, patch.timestamp) &&
                 Objects.equals(operations, patch.operations);
     }
 
     @Override
     @Generated("IntelliJ IDEA")
     public int hashCode() {
-        return Objects.hash(headers, operations);
+        return Objects.hash(headers, resource, timestamp, operations);
```

Comment:
I think so, too. We would have to introduce a sanity check on `resource` I guess since array types are not supported for that field